base/array.jl

```diff
@@ -711,8 +711,8 @@ function find(testf::Function, A::AbstractArray)
     # use a dynamic-length array to store the indexes, then copy to a non-padded
     # array for the return
     tmpI = Array(Int, 0)
-    for i = 1:length(A)
-        if testf(A[i])
+    for (i,j) = enumerate(eachindex(A))
+        if testf(A[j])
```

Comment:
`for (i, j) in enumerate(A)` is not @meggart wrote. He wrote `for (i, j) in enumerate(eachindex(A))` which seems like it would work.